spec/factories/responses.rb

```diff
@@ -1,65 +1,169 @@
+# frozen_string_literal: true
+
+# ResponseFactoryHelper builds a response tree based on answer_values
+# About the answer_values format:
+# The value of a select multiple or multilevel select answer is an array.
+#   For example, %w[Dog Cat] or %w[Cat] or %w[Plant Oak] or %w[Plant]
+# All other answer types are represented by an integer, string, or object as appropriate
+# A non repeat answer group of answers is an array of answer values
+# A repeat group is a hash with the key :repeating whose value is an array of answer group arrays
+#   e.g. {repeating: [ [1, "A"], [2, "B"] ] } where the group has an integer question and a text question
+#
+# Example answer_values with nested repeat groups:
+# root AnswerGroup    [
+# AnswerGroup          [
+# Answer                1
+#                      ],
+# Answer               create(:media_image),
+# Answer               %w[Plant Oak],
+# Outer rpt grp set   {repeating: [
+# Outer grp instance1   [
+# Answer                 2,
+# Inner rpt grp set       {repeating: [
+# Inner rpt grp instance1   [
+# Answer                      "a",
+# Answer                      10
+#                           ],
+# Inner rpt grp instance2   [
+# Answer                      "b",
+# Answer                      11
+#                           ]
+#                        ]}
+#                       ]
+#                     ]}
+#                    ]
```

Comment:
Nice big juicy comment!

Reply:
Should we put something about blank string vs. nil? Technically nil is a valid answer value and just means no object gets created? So nil implies irrelevant while "" implies relevant but blank?